### PR TITLE
Remove policyMemberList as part of personalDetails migration

### PR DESCRIPTION
**Pull request by puneetlath**
### Details

This PR updates the PersonalDetailsByAccountID migration to remove the old policyMemberList data from Onyx if it is found. We have updated the back-end to stop sending data to this Onyx key, so now we want to remove it from any clients that have it locally. 

### Fixed Issues

$ https://github.com/Expensify/App/issues/21454

### Tests

I added automated tests to ensure check that the data gets removed. To test manually, you can: 
1. In the JS console do `Onyx.set('policyMemberList_123', {blah: 1});`
2. Check policyMemberList_123 in application storage and make sure you see what you set
3. Refresh the page
4. Make sure that policyMemberList_123 is now empty in application storage

- [x] Verify that no errors appear in the JS console

### Offline tests

Same tests work online or offline. 

### QA Steps

There is nothing in particular to QA for this PR as there are no user-facing changes. Regular regression testing that the app is working as normal should be enough. 

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

</details>

<details>
<summary>Desktop</summary>

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

**Comment by puneetlath** [comment]
@s77rt maybe you can help me out here. I feel like I'm missing something obvious. 

When I do what is in this PR and I log the output, it shows that `oldPolicyMembers` is empty, even though I have multiple `policyMemberList_` collections in Onyx. 

image

If I change the constant on line 13 to be `policyMembers_` then it does find that collection, but for some reason when I make it `policyMemberList_` it doesn't find it. Any ideas what I'm doing wrong? 

image

**Comment by s77rt** [comment]
@puneetlath I think this has to do with [Onyx initialisation](https://github.com/Expensify/App/blob/27d657e5c08bfffccc141f90686848e26521a0e2/src/setup/index.js#L24C1-L43). Onyx only works with keys that are given at setup, we supply [ONYXKEYS](https://github.com/Expensify/App/blob/main/src/ONYXKEYS.js) and but this list does not contain `policyMemberList_`. Just adding that should fix the issue.

```js
POLICY_MEMBER_LIST: 'policyMemberList_',
```

**Comment by puneetlath** [comment]
Ahh interesting! Why does personalDetails work then? It's not in ONYXKEYS either.

**Comment by puneetlath** [comment]
Ok @s77rt I haven't had a chance to do the checklist and I need to head out for the day. But this is ready for review whenever you have a chance!

**Comment by s77rt** [comment]
> Why does personalDetails work then? It's not in ONYXKEYS either

I think this applies to collections only, `personalDetails` is not a collection, that key can be be found directly but when we search for `policyMemberList_` we mean to check for a collection and not for the literal key `policyMemberList_`. Onyx differentiate between a literal key and a collection key by use of [isCollectionKey](https://github.com/Expensify/react-native-onyx/blob/c25cec31b44adeb3d1af1fc518a592ecbed2390f/lib/Onyx.js#L144C11-L144C11) function which looks into `ONYXKEYS.COLLECTION` if the key is there then it's a collection otherwise a literal key.

- We used `policyMemberList_` which is a collection key but since it was not on `ONYXKEYS.COLLECTION` Onyx was searching for the literal `policyMemberList_` key instead which does not exist.
- `personalDetails` is a literal key. Onyx will search for that literal key anyway.


So in short, unless you tell Onyx that a key is a collection key it will treat it as a literal key.

**Comment by s77rt** [comment]
I will review asap

**Comment by puneetlath** [comment]
Thanks for the explanation @s77rt, that makes sense!

The update to stop sending policyMemberList data from the back-end is live. So it would be great to merge this part of the migration today if we can.

File: src/libs/migrations/PersonalDetailsByAccountID.js [hunk 1]
```diff
@@ -59,155 +75,165 @@ function getDeprecatedPersonalDetailsFromOnyx() {
...
-                        newReportAction.accountID = accountID;
-                    } else {
-                        Log.info(`[Migrate Onyx] PersonalDetailsByAccountID migration: removing reportAction ${reportActionID} because accountID not found`);
+            // We migrate reportActions to have the new accountID-based data if they don't already.
+            // If we are not able to get the accountID from personalDetails for some reason, we will just clear the reportAction
+            // and let it be fetched from the API next time they open the report and scroll to that action.
+            // We do this because we know the reportAction from the API will include the needed accountID data.
+            if (!oldReportActions) {
+                Log.info('[Migrate Onyx] Skipped PersonalDetailsByAccountID reportActions migration because there were no reportActions');
+            } else {
```
Comment (s77rt) [inline src/libs/migrations/PersonalDetailsByAccountID.js:95]: NAB but can we flatten this? No need for the else condition. Or just remove the log that we are skipping the migration (we don't have a similar log for oldPolicyMemberList being empty)

(this is just to clear the diff, it's hard to read but I assume nothing changed in the below code)

**Review by s77rt — COMMENTED** [review]
Working well, just a minor request to clear the code

File: src/libs/migrations/PersonalDetailsByAccountID.js (hunk 1 above)
Comment (puneetlath) [reply src/libs/migrations/PersonalDetailsByAccountID.js:95]: Hmm, yeah that log doesn't really do much for us. I'm going to remove it.

Comment (puneetlath) [reply src/libs/migrations/PersonalDetailsByAccountID.js]: Ok I updated it. Will be easier to review if you choose the "remove whitespace" option.

Comment (s77rt) [reply src/libs/migrations/PersonalDetailsByAccountID.js]: Thank you!

**Review by s77rt — APPROVED** [review]
LGTM! :rocket:

**Comment by puneetlath** [comment]
Nice! @arosiclair or @Beamanator if one of y'all has a chance to approve/merge today, that would be much appreciated!

**Review by Beamanator — APPROVED** [review]
LGTM!
